src/globus_sdk/services/timer/data.py

```diff
@@ -0,0 +1,113 @@
+from globus_sdk.utils import PayloadWrapper, slash_join
+
+
+class TimerJob(PayloadWrapper):
+    r"""
+    Class for specifying parameters used to create a job in the Timer service. Used as
+    the ``data`` argument in
+    :meth:`create_job <globus_sdk.TimerClient.create_job>`.
+
+    Timer operates through the `Globus Automate API
+    <https://docs.globus.org/globus-automation-services/>`_. Crucially, the
+    ``callback_url`` parameter should always be the URL used to run an action provider.
+    Currently the only supported action provider for this is for Transfer. Thus, users
+    should generally only use the :meth:`~from_transfer_data` method here. Any other
+    usage is meant for internal purposes; proceed with caution!
+
+
+    :param callback_url: URL for the action which the Timer job will use.
+    :type callback_url: str
+    :param callback_body: JSON data which Timer will send to the Action Provider on
+        each invocation
+    :param start: The datetime at which to start the Timer job.
+    :type start: datetime.datetime or str
+    :param interval: The interval at which the Timer job should recur.
+    :type interval: datetime.timedelta
+    :type callback_body: dict
+    :param name: A (not necessarily unique) name to identify this job in Timer
+    :type name: str, optional
+    :param stop_after: A date after which the Timer job will stop running
+    :type stop_after: datetime.datetime, optional
+    :param stop_after_n: A number of executions after which the Timer job will stop
+    :type stop_after_n: int
+    :param scope: The scope Timer will need in order to authorize against the Action
+        Provider specified in the callback URL.
+    :type scope: str, optional
+
+    .. automethodlist:: globus_sdk.TimerJob
+    """
+
+    def __init__(
+        self,
+        callback_url: str,
+        callback_body: Dict[str, Any],
+        start: Union[datetime.datetime, str],
+        interval: Union[datetime.timedelta, int, None],
+        *,
+        name: Optional[str] = None,
+        stop_after: Optional[datetime.datetime] = None,
+        stop_after_n: Optional[int] = None,
+        scope: Optional[str] = None,
+    ) -> None:
+        super().__init__()
+        self["callback_url"] = callback_url
+        self["callback_body"] = callback_body
+        if isinstance(start, datetime.datetime):
+            self["start"] = start.isoformat()
+        else:
+            self["start"] = start
+        if isinstance(interval, datetime.timedelta):
+            self["interval"] = int(interval.total_seconds())
+        else:
+            self["interval"] = interval
+        if name is not None:
+            self["name"] = name
+        if stop_after is not None:
+            self["stop_after"] = stop_after.isoformat()
+        if stop_after_n is not None:
+            self["stop_after_n"] = stop_after_n
+        if scope is not None:
+            self["scope"] = scope
+
+    @classmethod
+    def from_transfer_data(
```

Comment:
I'm leaving my comments below because they'll show my thought process on this a bit, but I'd also like to offer an alternative way forward. I have questions/concerns about handling environments nicely, and I want various changes which will be tedious/laborious.

If this doesn't seem like a quick solve and we want to unblock things, we can remove this helper and add the following method:

```python
@classmethod
def get_transfer_action_url(
    cls, *, environment: Optional[str] = None
) -> str:
    return slash_join(
        get_service_url("actions", environment=environment), "transfer/transfer/run"
    )
```

plus an example added to the `docs/examples/` dir which shows how to use this to construct a timer task.

That will set us up to implement something that does this assembly step but we can think about how to handle env names being set at runtime without blocking the rest of the work.

This isn't the only way forward. If there's a way of refactoring this which satisfies my desire to support setting `environment` at runtime, I'm okay with keeping this more sophisticated helper.